<launch>
    <arg name="enable_ekf" default="true"/>
    <arg name="is_sim" default="false"/>
    <arg name="rvizconfig" default="$(find robotics_bringup)/rviz/sensors.rviz"/>

    <group if="$(arg enable_ekf)">
        <node name="robot_localization" pkg="robot_localization" type="ekf_localization_node">
            <rosparam command="load" file="$(find robotics_bringup)/config/localization.yaml"/>

            <remap from="odometry/filtered" to="odom"/>
        </node>
        <node name="odom_covariance_node" pkg="robotics_bringup" type="odom_covariance_node">
            <param name="input_odom_topic" type="string" value="scout/odom"/>
            <param name="output_odom_topic" type="string" value="scout/odom_covar"/>
        </node>
    </group>

    <group unless="$(arg is_sim)">
        <include file="$(find urg_node)/launch/urg_lidar.launch"/>
        <include file="$(find realsense2_camera)/launch/rs_camera.launch">
            <arg name="camera" value="realsense"/>
            <arg name="align_depth" value="true"/>
            <arg name="enable_gyro" value="true"/>
            <arg name="enable_accel" value="true"/>
            <arg name="unite_imu_method" value="linear_interpolation"/>
            <arg name="publish_tf" value="false"/>
            <arg name="publish_odom_tf" value="false"/>
        </include>
    </group>

    <include file="$(find robotics_description)/launch/description.launch"/>

    <node name="scout_base_node" pkg="scout_base" type="scout_base_node">
        <param name="is_scout_mini" type="bool" value="true"/>
        <param name="simulated_robot" type="bool" value="$(arg is_sim)"/>
        <param name="odom_frame" type="string" value="odom"/>
        <param name="base_frame" type="string" value="base_link"/>
        <param name="odom_topic_name" type="string" value="scout/odom"/>
        <param if="$(arg enable_ekf)" name="pub_tf" type="bool" value="false"/>
        <param unless="$(arg enable_ekf)" name="pub_tf" type="bool" value="true"/>

        <remap unless="$(arg enable_ekf)" from="scout/odom" to="odom"/>
    </node>
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)"/>
</launch>
